components/board/fake/board.go

```diff
@@ -233,39 +238,51 @@ func (b *Board) SetPowerMode(ctx context.Context, mode pb.PowerMode, duration *t
 func (b *Board) StreamTicks(ctx context.Context, interrupts []board.DigitalInterrupt, ch chan board.Tick,
 	extra map[string]interface{},
 ) error {
-	for _, i := range interrupts {
-		name := i.Name()
-		d, ok := b.Digitals[name]
+	for _, di := range interrupts {
+		_, ok := b.Digitals[di.Name()]
 		if !ok {
-			return fmt.Errorf("could not find digital interrupt: %s", name)
-		}
-		select {
-		case <-ctx.Done():
-			return ctx.Err()
-		default:
-			// Keep going
-		}
-		// Get a random bool for the high tick value.
-		// linter complans about security but we don't care if someone
-		// can predict if the fake interrupts will be high or low.
-		//nolint:gosec
-		randBool := rand.Int()%2 == 0
-		select {
-		case ch <- board.Tick{Name: d.conf.Name, High: randBool, TimestampNanosec: uint64(time.Now().Unix())}:
-		default:
-			// if nothing is listening to the channel just do nothing.
+			return fmt.Errorf("could not find digital interrupt: %s", di.Name())
```

Comment:
:+1: Good call here, making sure all the interrupts are valid before creating any background goroutines!